Makefile

```diff
@@ -79,8 +80,16 @@ ifeq (${MBED_TOOLCHAIN},)
   MBED_TOOLCHAIN:=${DEFAULT_TOOLCHAIN}
 endif
 
-# Specifies the path to the directory containing build output files
-MBED_BUILD_DIR:=./BUILD/${MBED_TARGET}/${MBED_TOOLCHAIN}
+# Specifies the path to the directory containing build output files.
+# When building with mbed-cli 1.8.x, build profiles are appended to
+# target build directory names in upper case letters.
+# Example:
+# ./BUILD/UBLOX_EVK_ODIN_W2/GCC_ARM/ becomes
+# ./BUILD/UBLOX_EVK_ODIN_W2/GCC_ARM-RELEASE/
+MBED_TOOLCHAIN_EXTENSION=$(shell basename ${BUILD_PROFILE} .json | awk '{print toupper($$0)}')
+BOOTLDR_TOOLCHAIN_EXTENSION=$(shell basename ${BOOTLOADER_BUILD_PROFILE} .json | awk '{print toupper($$0)}')
+MBED_BUILD_DIR:=./BUILD/${MBED_TARGET}/${MBED_TOOLCHAIN}-${MBED_TOOLCHAIN_EXTENSION}
```

Comment:
I suggest making the hyphen "-" between MBED_TOOLCHAIN and MBED_TOOLCHAIN_EXTENSION a part of the MBED_TOOLCHAIN_EXTENSION value.  That way, 
 MBED_TOOLCHAIN_EXTENSION could be an empty string and the MBED_BUILD_DIR path would still be a valid path.